test/rhash/test_hash.c

```diff
@@ -804,6 +804,21 @@ static void test_hash_dreamcast_cue()
 
 /* ========================================================================= */
 
+static void test_hash_gamecube()
+{
+  size_t image_size;
+  uint8_t* image = generate_gamecube_iso(32, &image_size);
+  char hash[33];
+  int result = rc_hash_generate_from_buffer(hash, RC_CONSOLE_GAMECUBE, image, image_size);
```

Comment:
gamecube should never be buffered. mock the file and call `rc_hash_generate_from_file`

```
  const char* filename = "test.iso";
  mock_file(0, filename, image, image_size);
  int result = rc_hash_generate_from_file(hash, RC_CONSOLE_GAMECUBE, filename);
```

Reply:
Okay, but I'm still trying to wrap my head around quite how to implement compression. If it's not in rcheevos then my inclination is to pass the decompressed buffer in, but I know you've mentioned CD readers in the past so I'll look into that.